tests/tasks/manage_test_interface.yml

```
@@ -1,10 +1,12 @@
 # SPDX-License-Identifier: BSD-3-Clause
 ---
-- fail:
+- name: Ensure state not in ["present", "absent"]
```

```suggestion
- name: Ensure state in ["present", "absent"]
```

This is the fail module, so this ensures that the state is present or absent. Maybe assert would be better here to make this clear.

---

Ah, you are right. A good catch. Thank you, @tyll!

If we use `assert` here, may I replace the `fail` module with `assert` as follows?
```
-- name: Ensure state not in ["present", "absent"]
-  fail:
-    msg: "state needs to be present or absent, not '{{ state }}'"
-  when: state not in ["present", "absent"]
+- name: Ensure state in ["present", "absent"]
+  assert:
+    that: state in ["present", "absent"]
```
Or shall I just add the correct `name`?

---

The assert looks correct to, I am not sure if there are any pitfalls with assert in Ansible that might make this incorrect.